core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -1462,7 +1618,7 @@ private ExpressionNode parseTimestamp(GenericLexer lexer, CharSequence tok) thro
         return null;
     }
 
-    private QueryModel parseWith(GenericLexer lexer, WithClauseModel wcm, LowerCaseCharSequenceObjHashMap<WithClauseModel> withClauses) throws SqlException {
+    private QueryModel parseWith(GenericLexer lexer, WithClauseModel wcm, QueryWithClauseModel withClauses) throws SqlException {
```

Comment:
this is contentious :)

`QueryWithClauseModel` is a simple wrapper for `LowerCaseCharSequenceObjHashMap`. All methods on `QueryWithClauseModel` could be made static to save a wrap?

Reply:
ok